### ISSUE TITLE
router handlers/endpoints should include a context

### ISSUE THREAD

**Issue by mvdan** [vocdoni/multirpc#2]
So that the context can be cancelled when a client disconnects, or so that an entire router can have a default timeout for all requests to avoid leaking goroutines or DoS attacks with clients keeping connections open for hours.

**Comment by p4u** [comment]
Yes, that makes sense and we should do it ASAP. Is it fine if I assign it to you? Else it might be done also by @jordipainan or myself.

**Comment by mvdan** [comment]
Sure - I think we should do it at the same time as reviewing the interfaces. Do you prefer if I propose changes directly as a PR, or if I explain the changes I want to make in written text or on a call?

**Comment by p4u** [comment]
Well, if they are not very structural changes a PR would be the best. If you want to discuss some general design, then lets talk :)

**Comment by mvdan** [comment]
Yep, I do mean structural changes, mainly on the interfaces :)

**Comment by p4u** [comment]
Ok, so lets arrange a meeting.
